test/integration/targets/docker_network/tasks/tests/ipam.yml

```diff
@@ -0,0 +1,172 @@
+---
+- name: Registering network names
+  set_fact:
+    nname_ipam_0: "{{ name_prefix ~ '-network-ipam-0' }}"
+    nname_ipam_1: "{{ name_prefix ~ '-network-ipam-1' }}"
+    nname_ipam_2: "{{ name_prefix ~ '-network-ipam-2' }}"
+    nname_ipam_3: "{{ name_prefix ~ '-network-ipam-3' }}"
+
+- name: Registering network names
+  set_fact:
+    dnetworks: "{{ dnetworks }} + [nname_ipam_0, nname_ipam_1, nname_ipam_2, nname_ipam_3]"
+
+
+#################### network-ipam-0 deprecated ####################
+
+- name: Create network with ipam_config and deprecated ipam_options
+  docker_network:
+    name: "{{ nname_ipam_0 }}"
+    ipam_options:
+      subnet: 172.3.29.0/24
+    ipam_config:
+      - subnet: 172.3.29.0/24
+  register: network
+  ignore_errors: yes
+
+- assert:
+    that:
+      - network is failed
+      - network.msg == "Only one of `ipam_options` and `ipam_config` can be defined."
+
+- name: Create network with deprecated custom IPAM config
+  docker_network:
+    name: "{{ nname_ipam_0 }}"
+    ipam_options:
+      subnet: 172.3.29.0/24
+  register: network
+
+- assert:
+    that:
+      - network is changed
+
+- name: Change subnet of network with deprecated custom IPAM config
+  docker_network:
+    name: "{{ nname_ipam_0 }}"
+    ipam_options:
+      subnet: 172.3.30.0/24
+  register: network
+  diff: yes
+
+- assert:
+    that:
+      - network is changed
+      - network['diff'] | length == 1
+      - network['diff'][0] == "ipam_config[0].subnet"
+
+
+#################### network-ipam-1 ####################
+- name: Create network with custom IPAM config
+  docker_network:
+    name: "{{ nname_ipam_1 }}"
+    ipam_config:
+      - subnet: 172.3.27.0/24
+        gateway: 172.3.27.1
+        iprange: 192.168.1.0/24
```

Comment:
The IP range doesn't lie in the subset. I don't think that's a good idea. It could even be that future versions of docker might fail because of this.

Reply:
I copied that from the earlier example. Fixing.